dre-service/src/main/java/io/cdap/re/RulesDB.java

```diff
@@ -404,7 +405,7 @@ void addRuleToRulebook(NamespacedId rulebookNamespacedId, NamespacedId ruleNames
     Collection<Field<?>> rulebookFields = new ArrayList<>(rulebookKeyFields);
     rulebookFields.add(Fields.longField(UPDATED_COL, getCurrentTime()));
     rulebookFields.add(Fields.longField(VERSION_COL, rulebookRow.getLong(VERSION_COL)));
-    rulebookFields.add(Fields.stringField(RULES_COL, GSON.toJson(rules)));
+    rulebookFields.add(Fields.stringField(RULES_COL, Joiner.on(",").join(rules)));
```

Comment:
It's better to use json than a comma separated list, as it doesn't rely on rules never having a comma in them.

This seems to indicate that whatever reads this field is not using Gson to read the field as a List<String>.

Reply:
The problem that we encountered was when [exporting a rulebook](https://github.com/cdap-solutions/dre/blob/5350a06b0dd5863b50e1f93f0ecfeb8a464081bf/dre-service/src/main/java/io/cdap/re/RulesDB.java#L449) and [importing a rulebook](https://github.com/cdap-solutions/dre/blob/5350a06b0dd5863b50e1f93f0ecfeb8a464081bf/dre-service/src/main/java/io/cdap/re/YARERulebookService.java#L221). Exporting the rulebook worked correctly, but we got a [RulebookCompileException](https://github.com/cdap-solutions/dre/blob/develop/dre-api/src/main/java/io/cdap/re/RulebookCompileException.java). There seems to be a format that is expected when importing and parsing a rulebook that is failing when we return a list of rules instead of a comma separated list.

Reply:
ok let's do this for now then

Reply:
Updated in the latest commit